### PR TITLE
chore: add setFullScreen to component's ref

### PR DESCRIPTION
**Pull request by seyedmostafahasani**
…eenPlayer & dismissFullscreenPlayer

## Summary
I added `setFullScreen` to handle fullScreen mode without controls.

### Motivation
The user can use it to handle fullscreen mode with custom controls, specifically refactoring the codebase and fix this [issue](https://github.com/TheWidlarzGroup/react-native-video/issues/3732)

### Changes
I added a function called `setFullScreen` and deprecated the functions `presentFullscreenPlayer` and `dismissFullscreenPlayer`.

## Test plan

### CODE REVIEW

**Review by freeboub — CHANGES_REQUESTED** [review]
We cannot remove existing APIs on 6.X branch, I don't understand why you need this change. Can you please clarify ?

**Comment by seyedmostafahasani** [comment]
I think we can implement full-screen mode with just this function. We don't need two functions to handle it. Additionally, if we remove these functions, we will eliminate an unnecessary useState from the video component, which prevents re-rendering. Specifically, we will also fix another issue in the library, #3732. The main point is that we handle this logic with just one function. What is your opinion?

**Comment by seyedmostafahasani** [comment]
@freeboub what do you think about my points?

**Comment by freeboub** [comment]
@seyedmostafahasani We cannot change api on V6.X, if we integrate such api change, we will need to bump to V7 and all apps using the package will have to upgrade their integration. as the team have a lot of api changes in mind, we try to keep api interoperability as much as possible.

Here is a sample: 
https://github.com/TheWidlarzGroup/react-native-video/pull/3867/files#diff-e2131d0631cdc1ddf0ba36827a27b86adf8e5b8ed162a280a53c6219f4df5a5f

I just move few props from the root of video to the source object, to avoid upgrade issue, I comment in doc that the old api is deprecated, and point to the new way to proceed.

It will allow easy upgrade for apps, we keep in mind that old api (deprecated) can be removed on V7.
Can you please do the same thing please ?
I think merging the 2 apis make sense, but we need to keep compatibility of old api some time

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 1]
```diff
@@ -2184,18 +2184,14 @@ public boolean getPreventsDisplaySleepDuringVideoPlayback() {
         return preventsDisplaySleepDuringVideoPlayback;
     }
 
-    private void updateFullScreenButtonVisbility() {
+    private void updateFullScreenButtonVisibility() {
         if (playerControlView != null) {
             final ImageButton fullScreenButton = playerControlView.findViewById(R.id.exo_fullscreen);
-            if (controls) {
-                //Handling the fullScreenButton click event
-                if (isFullscreen && fullScreenPlayerView != null && !fullScreenPlayerView.isShowing()) {
```
Comment (freeboub) [inline android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:2192]: why did you remove this section ?
Please check git history, but I think it was fixing a problem !

**Review by freeboub — CHANGES_REQUESTED** [review]

**Comment by seyedmostafahasani** [comment]
> @seyedmostafahasani We cannot change api on V6.X, if we integrate such api change, we will need to bump to V7 and all apps using the package will have to upgrade their integration. as the team have a lot of api changes in mind, we try to keep api interoperability as much as possible.
> 
> Here is a sample: https://github.com/TheWidlarzGroup/react-native-video/pull/3867/files#diff-e2131d0631cdc1ddf0ba36827a27b86adf8e5b8ed162a280a53c6219f4df5a5f
> 
> I just move few props from the root of video to the source object, to avoid upgrade issue, I comment in doc that the old api is deprecated, and point to the new way to proceed.
> 
> It will allow easy upgrade for apps, we keep in mind that old api (deprecated) can be removed on V7. Can you please do the same thing please ? I think merging the 2 apis make sense, but we need to keep compatibility of old api some time

I completely agree with you about deprecating the old API and planning to remove it in V7. I will add deprecation notices for both `presentFullscreenPlayer` and `dismissFullscreenPlayer`. This approach ensures backward compatibility while allowing a smooth transition for users until V7, where we can fully remove the old API.

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 2]
```diff
@@ -2219,7 +2219,7 @@ public void setFullscreen(boolean fullscreen) {
         WindowInsetsControllerCompat controller = new WindowInsetsControllerCompat(window, window.getDecorView());
         if (isFullscreen) {
             eventEmitter.fullscreenWillPresent();
-            if (controls && fullScreenPlayerView != null) {
```
Comment (freeboub) [inline android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:2222]: not sure why you did that, but in the absolute, it makes sense.
BTW, I think this "control" test is not here as if controls == false, fullScreenPlayerView shall also be null

File: docs/pages/component/methods.mdx [hunk 3]
```diff
@@ -12,26 +12,44 @@ This page shows the list of available methods
 
 Take the player out of fullscreen mode.
 
-### `pause`
```
Comment (freeboub) [inline docs/pages/component/methods.mdx:15]: I don't understand why you move pause, It is the first one in alphabetical order !

**Comment by freeboub** [comment]
@seyedmostafahasani Thank you for the fix,
last remaining point, can you please remove deprecated functions in the sample please ? :)

following code shall be simplified:
 
```javascript
  toggleDecoration() {
    this.setState({decoration: !this.state.decoration});
    if (this.state.decoration) {
      this.video?.dismissFullscreenPlayer();
    } else {
      this.video?.presentFullscreenPlayer();
    }
  }
```

File: docs/pages/component/methods.mdx [hunk 4]
```diff
@@ -32,6 +35,21 @@ On iOS, this displays the video in a fullscreen view controller with controls.
 
 On Android, this puts the navigation controls in fullscreen mode. It is not a complete fullscreen implementation, so you will still need to apply a style that makes the width and height match your screen dimensions to get a fullscreen video.
 
+> [!WARNING]
+> deprecated, use setfullScreen method instead
+
+### `setFullScreen`
```
Comment (freeboub) [inline docs/pages/component/methods.mdx]: There is an ordering issue here !

**Review by freeboub — CHANGES_REQUESTED** [review]

**Comment by seyedmostafahasani** [comment]
@freeboub I removed the controls condition in the `updateFullScreenButtonVisibility` function. When the user sets `controls={false}`, the fullScreen button disappears in fullScreen mode, requiring the user to press the physical back button to exit fullScreen mode, which may cause confusion. What is your opinion?

File: examples/basic/src/VideoPlayer.tsx [hunk 5]
```diff
@@ -431,9 +431,9 @@ class VideoPlayer extends Component {
   toggleDecoration() {
     this.setState({decoration: !this.state.decoration});
     if (this.state.decoration) {
```
Comment (freeboub) [inline examples/basic/src/VideoPlayer.tsx]: this.video?.setFullScreen(!this.state.decoration);
would be cleaner! The goal is to remove lines here :)

**Comment by YangJonghun** [comment]
I don't use native controls and fullscreen in this library, but I'm not sure if this breaking change is really necessary in this library.
It's not a duplicate function, so it doesn't have much effect on user DX as long as user pass an argument. Also, It's easy to remove useState and keep the existing API, so why we make breaking change?

**Comment by freeboub** [comment]
It is not a breaking change as previous api are still working! It is just a preparation for api clean up ! (Let say for V7)

**Comment by YangJonghun** [comment]
@freeboub 
oh..then I misunderstood.
(If possible, it would be nice to have something like a new-arch only prop that works synchronously. `ref.fullscreen = true`)

**Comment by freeboub** [comment]
Yes I agree, Fullscreen management is one of the next challenge I think...

**Review by KrzysztofMoch — APPROVED** [review]
LGTM

**Review by freeboub — APPROVED** [review]

**Pull request merged**